pkgs/tools/misc/qrscan/default.nix

```
@@ -0,0 +1,40 @@
+{ lib, rustPlatform, fetchFromGitHub }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "qrscan";
+  version = "0.1.9";
+
+  src = fetchFromGitHub {
+    owner = "sayanarijit";
+    repo = "qrscan";
+    rev = "v${version}";
+    hash = "sha256-nAUZUE7NppsCAV8UyR8+OkikT4nJtnamsSVeyNz21EQ=";
+  };
+
+  nativeBuildInputs = [
+    rustPlatform.bindgenHook
+  ];
+
+  cargoPatches = [
+    # half 2.3.1 requires rustc 1.70 or newer, but we're using 1.69
+    ./cargo.lock.patch
+  ];
+
+  cargoHash = "sha256-tjv6BzVeMIFC8+9CKxJ8ZpFGvMRCdfVKt4KcCkwQ04U=";
```

rustc v1.70 is landed in nixos-unstable, so this isn't needed anymore :)

```suggestion
  cargoHash = "sha256-P40IwFRtEQp6BGRgmt1x3UXtAKtWaMjR3kqhYq+p7wQ=";
```

---

Cool...